### PR TITLE
Add chapter on Yocto caching

### PR DESCRIPTION
**Pull request by tobsan**
The chapter describes yocto cache in general, and how it works over HTTP
and NFS.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: docs/articles/infrastructure/ci-cd/yocto-cache.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+to be set: ``SOURCE_MIRROR_URL`` and ``SSTATE_MIRRORS``. The first one points to
+where source packages can be found, the second points to the cached binary
+builds.
+
+.. code-block: bash
+    :caption: site.conf
+
+    INHERIT += "own-mirrors"
+    SOURCE_MIRROR_URL ?= "http://example.org/yocto-cache/downloads/"
+    SSTATE_MIRRORS ?= "file://.* http://example.org/yocto-cache/sstate-cache/PATH"
```
Comment (rpannek) [inline docs/articles/infrastructure/ci-cd/yocto-cache.rst]: What does the `file://.*` do here?

File: docs/articles/infrastructure/ci-cd/yocto-cache.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        }
+    }
+
+When one updates the cache, this means writing to /var/www/yocto-cache in some
+way.
+
+Cache over NFS
+--------------
+Setting up a cache over NFS requires more setup on the server-side, but on the
+other hand, the cache can be kept private from the outside world.
```
Comment (rpannek) [inline docs/articles/infrastructure/ci-cd/yocto-cache.rst]: You can keep the http also either behind a VPN or even easier with a basic-auth password in the URL: `https://user:password@example.com/yocto-cache/` if you use https the URL is send encrypted so no problem having the password there either.

I'd remove the statement about having it private and would rather add something about everyone being able to update the cache or something like that.

**Review by rpannek — CHANGES_REQUESTED** [review]

File: docs/articles/infrastructure/ci-cd/yocto-cache.rst (hunk 1 above)
Comment (tobsan) [reply docs/articles/infrastructure/ci-cd/yocto-cache.rst]: It's from the guide on how to set these things up actually. Not sure it is needed actually, but just copied from the existing `site.conf` in pelux-manifests.

File: docs/articles/infrastructure/ci-cd/yocto-cache.rst (hunk 2 above)
Comment (tobsan) [reply docs/articles/infrastructure/ci-cd/yocto-cache.rst]: Fair point!

**Comment by tobsan** [comment]
Updated!

File: docs/articles/infrastructure/ci-cd/yocto-cache.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
+Yocto cache
+===========
```
Comment (breakreturn) [inline docs/articles/infrastructure/ci-cd/yocto-cache.rst]: This looks like a how-to basically. I think the title and file name should reflect that something is to be done, e.g. set up a yocto cache, and there should be a label that indicates what type of article it is (see recent additions of labels).

**Review by breakreturn — CHANGES_REQUESTED** [review]

File: docs/articles/infrastructure/ci-cd/yocto-cache.rst (hunk 1 above)
Comment (jeena) [reply docs/articles/infrastructure/ci-cd/yocto-cache.rst]: Ah yeah, I remember now putting it there and when I tested without then it wouldn't work for some reason.

**Review by rpannek — APPROVED** [review]

File: docs/articles/infrastructure/ci-cd/yocto-cache.rst (hunk 3 above)
Comment (tobsan) [reply docs/articles/infrastructure/ci-cd/yocto-cache.rst]: Fair point.

**Comment by tobsan** [comment]
@breakreturn Updated, please review.

**Review by breakreturn — APPROVED** [review]

**Pull request merged**